src/amo/components/Addon/index.js

```diff
@@ -629,7 +629,12 @@ export function mapStateToProps(state, ownProps) {
   let installedAddon = {};
 
   if (addon) {
-    addonsByAuthors = state.addonsByAuthors.byAddonSlug[addon.slug];
+    const ids = state.addonsByAuthors.byAddonSlug[addon.slug];
+    if (ids) {
+      addonsByAuthors = ids.map((id) => {
```

Comment:
It would be wise to move this to a selector function defined in the `addonsByAuthors` reducer because that makes future refactors of the state a lot easier.